flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobAutoScalerImpl.java

```diff
@@ -159,19 +161,24 @@ private void runScalingLogic(Context ctx, AutoscalerFlinkMetrics autoscalerMetri
             throws Exception {
 
         var collectedMetrics = metricsCollector.updateMetrics(ctx, stateStore);
+        var jobTopology = collectedMetrics.getJobTopology();
 
         if (collectedMetrics.getMetricHistory().isEmpty()) {
             return;
         }
         LOG.debug("Collected metrics: {}", collectedMetrics);
 
-        var evaluatedMetrics = evaluator.evaluate(ctx.getConfiguration(), collectedMetrics);
+        var now = clock.instant();
+        // Scaling tracking data contains previous restart times that are taken into account
+        var scalingTracking = getTrimmedScalingTracking(stateStore, ctx, now);
+        var evaluatedMetrics =
+                evaluator.evaluate(ctx.getConfiguration(), collectedMetrics, scalingTracking);
```

Comment:
I don't think this works with the current metrics scoping since it would lead to duplicating the restart time per vertex and we are striving to minimizing the size of the config map. Also, instead of just checking one tracking entry, should we instead iterate over all records of this metric across all vertices and take the maximum over that? Or just trust one observation? If we'll just use one observation, why would we need this data in every vertex and if we know it is supposed to be the same for all vertices, why store it at the vertex level?

Reply:
How do you propose to make the association between the rescaling decision and the collected metrics record? My understanding is that `updateMetrics` writes collected metrics irrespective from the rescalings, so we need to somehow decide which history entry to associate the restart time with. To be honest this feels like misusing the data structure not for its original purposes and making code harder to interpret.

Reply:
Sorry, small correction: I meant `evaluatedMetrics`, not `collectedMetrics`.

We don't need to associate the history entry with any of the metrics used for scaling. We would insert into the evaluated metrics map whatever the current determined rescale time is. This is similar to the MAX_PARALLELISM metric.

Reply:
The way the scalingHistory is currently structured, this would currently lead to duplicating the restart time unnecessarily per job vertex.